MIGRATION_GUIDE.md

```diff
@@ -4,6 +4,28 @@ This document is meant to help you migrate your Terraform config to the new newe
 describe deprecations or breaking changes and help you to change your configuration to keep the same (or similar) behavior
 across different versions.
 
+## v0.89.0 ➞ v0.90.0
+### tag_masking_policy_association resource changes
+Now the `tag_masking_policy_association` resource will only accept fully qualified names separated by dot `.` instead of pipe `|`.
```

Comment:
I think we should add a state upgrader to make it easy for the users (this is not much work and can be done in a separate PR - but before the release). WDYT?

Reply:
True, will work on it and push the next pr

Reply:
I think I'll leave this as it is, would have to write a few state upgrades and make changes to how IDs are saved/retrieved also in a few places, which makes me feel something could go wrong there. Also, forcing the migration to a more complex ID, but also a more correct one would most likely result in no migration after identifiers rework (because it's already a valid identifier. If we add some helpers field like `qualified_name` it will depend on the user when they would like to upgrade).